### PR TITLE
KAA-1371: Update Programming guide -> Your first Kaa application

### PR DESCRIPTION
**Pull request by nocs00**

### CODE REVIEW

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 1]
```diff
@@ -88,18 +88,16 @@ To create and upload the schemas, proceed as follows:
...
-1. Scroll down and use the **Upload from file** function to find the previously created `data-schema.json` file with the schema.
-Alternatively, you can use the [Schema Avro UI]({{root_url}}/Administration-guide/Tenants-and-applications-management/#avro-ui-forms) form to create the schema.
-1. Click **Upload**.
+    ![Data collection scheme](attach/new_data_schema2.png)
 1. Click **Add** at the top of the window.
-1. Repeat uploading and adding for the configuration schema (`configuration-schema.json`) scheme:
+1. Repeat uploading and adding for the configuration schema (from file `configuration-schema.json`):
 
-    ![Data collection scheme](attach/new_config_schema.png)
+    ![Data collection scheme](attach/new_config_schema2.png)
```
Comment (ikulikov) [inline doc/Programming-guide/Your-first-Kaa-application/index.md:101]: [Add configuration schema] ?

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 2]
```diff
@@ -580,6 +589,18 @@ To launch C application next steps should be performed.
 
 <div id="run-java" class="tab-pane fade" markdown="1" >
 
+To launch Java application next steps should be performed.
+
+1. Save application code into FirstKaaDemo.java file into `demo_app` directory.
```
Comment (ikulikov) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: Save application code into FirstKaaDemo.java file <b>in</b> `demo_app` directory.

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 3]
```diff
@@ -206,7 +204,15 @@ Before you start with C application code, some preparation is required.
 </div>
 
 <div id="prep-java" class="tab-pane fade" markdown="1" >
+Before you start with Java application code, some preparation is required.
 
+1. Install [Oracle JDK 8](http://www.oracle.com/technetwork/java/javase/downloads/index.html).
+
+        $ sudo add-apt-repository ppa:webupd8team/java
```
Comment (ikulikov) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: What about different OS, for ex. Windows.

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 4]
```diff
@@ -580,6 +589,18 @@ To launch C application next steps should be performed.
...
+
+1. Save application code into FirstKaaDemo.java file into `demo_app` directory.
+1. Navigate to `demo_app` directory.
+1. Build application.
+
+        javac -cp *.jar *.java
+
+1. Launch application.
+        
+        java -cp '.:./*' FirstKaaDemo
```
Comment (ikulikov) [inline doc/Programming-guide/Your-first-Kaa-application/index.md:599]: Please note that in Unix and Windows systems different path and file separator chars are used. For path: ':' and ';', for file '/' and '\' accordingly.

**Review by ikulikov — CHANGES_REQUESTED** [review]

File: doc/Programming-guide/Your-first-Kaa-application/index.md [hunk 5]
```diff
@@ -88,18 +88,16 @@ To create and upload the schemas, proceed as follows:
...
 1. Open the [admin UI]({{root_url}}Administration-guide/Tenants-and-applications-management) in your browser and log in as a tenant developer (default user/password in the sandbox: `devuser`/`devuser123`; use the correct credentials if you changed the default).
 1. Open the relevant **Log schemas** window (**Applications => My First Kaa Application => Schemas => Log**) and click **Add schema**.
-1. In the **Add log schema** window, enter the name and description of the new data collection schema.
+1. In the **Add log schema** window, enter the name and description of the new data collection schema and select CTL created before (from file `data-schema.json`).
 
-    ![Data collection scheme](attach/new_data_schema.png)
-1. Scroll down and use the **Upload from file** function to find the previously created `data-schema.json` file with the schema.
-Alternatively, you can use the [Schema Avro UI]({{root_url}}/Administration-guide/Tenants-and-applications-management/#avro-ui-forms) form to create the schema.
-1. Click **Upload**.
+    ![Data collection scheme](attach/new_data_schema2.png)
```
Comment (ikulikov) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: [Add log schema] ?

**Review by ikulikov — CHANGES_REQUESTED** [review]

File: doc/Programming-guide/Your-first-Kaa-application/index.md (hunk 1 above)
Comment (nocs00) [inline doc/Programming-guide/Your-first-Kaa-application/index.md]: fixed

**Pull request merged**